app/src/main/java/org/simple/clinic/remoteconfig/ConfigReader.kt

```diff
@@ -1,8 +1,11 @@
 package org.simple.clinic.remoteconfig
 
+import io.reactivex.Completable
+
 interface ConfigReader {
   fun string(name: String, default: String): String
   fun boolean(name: String, default: Boolean): Boolean
   fun double(name: String, default: Double): Double
   fun long(name: String, default: Long): Long
+  fun update(): Completable
```

Comment:
Adding an `update` method in a reader, looks out of place since the `ConfigReader` is a utility class to just access config.
Updating it should be a separate `interface` like `ConfigUpdate` or `ConfigSync`.
Wdyt?

Reply:
I intentionally added this to this interface because if we need to connect to a different remote config service, we only need to update in one place instead of multiple. Maybe a different name for this instead of `ConfigReader`?

There's also another option I can explore:

```kotlin
interface RemoteConfigService {
  fun reader(): ConfigReader
  fun update(): Completable
}

class FirebaseRemoteConfigService {
  fun reader() = FirebaseConfigReader()
  fun update(): Completable {
    // update remote config here
  }
}
```

What do you think?

Reply:
I understand the reason for making this change. I was only looking to separate the `ConfigReader` from the `update` operation itself.
Also, we were exposing `update()` to classes that only need to access config values.

The implementation suggested in you comment looks good. So this will mean that classes that need to read config will only be inject the `ConfigReader` and not `RemoteConfigService`.

Reply:
They will receive the service and retrieve the reader through the service. So when we switch do a different service, we will be forced to implement both.

I will go down this route and see where it goes.

Reply:
Done.